Kadabra-JS/src-api/kadabra/Concurrent.ts

```diff
@@ -0,0 +1,130 @@
+    fieldName: string,
+    initValue: string,
+    isStatic: boolean = true
+) {
+    const convert = convertPrimitive(type);
+    let atomicType: string = "java.util.concurrent.atomic.Atomic";
+    let atomicTypeSimple: string = "Atomic";
+
+    if (convert.isPrimitive) {
+        atomicType += convert.wrapper;
+        atomicTypeSimple += convert.wrapper;
+    } else {
+        atomicType += "Reference<" + type + ">";
+        atomicTypeSimple += "Reference<>";
+    }
+
+    const init = initValue || "";
+    const mods = ["private"];
+    if (isStatic) {
+        mods.push("static");
+    }
+    const field = $class.newField(
+        mods,
+        atomicType,
+        fieldName,
+        "new " + atomicTypeSimple + "(" + init + ")"
+    );
+    const reference = field.declarator + "." + field.name;
+    const name = field.name;
+    const get = reference + ".get()";
+    const set = function (value: string) {
+        return reference + ".set(" + value + ")";
+    };
+
+    return { reference, name, field, get, set };
+}
+
+export function NewThread(
+    $class: Class,
+    threadName: string = "thread",
+    adaptCode: string = ""
+) {
+    const names = getAPINames();
+
+    //Add thread field
+    const $thread = $class.newField(
+        ["private", "static"],
+        names.thread,
+        threadName,
+        "new " + names.thread + "()"
+    );
+    //And the method to execute
+    const threadMethod = $class.newMethod(
+        ["private", "static"],
+        "void",
+        threadName + "Method",
+        [],
+        [],
+        adaptCode
+    );
+
+    const name = $thread.name;
+    const reference = $thread.declarator + "." + name;
+    const start =
+        reference +
+        ".start(" +
+        threadMethod.declarator +
+        "::" +
+        threadMethod.name +
+        ")";
+    const stop = reference + ".terminate()";
+    const running = reference + ".isRunning()";
+
+    const setCode = (code: string) => {
+        threadMethod.body.replaceWith(code);
+    }; //allows adaptation code rewritting
+    return { start, stop, running, setCode, name, reference };
+}
+
+export function NewChannel(
+    keyTypeI: string,
+    valueTypeI: string,
+    $class: Class,
+    capacity: string,
+    channelName = "channel",
+    isStatic = true
+) {
+    let convert = convertPrimitive(keyTypeI);
+    const keyType: string = convert.wrapper;
+    convert = convertPrimitive(valueTypeI);
+    const valueType: string = convert.wrapper;
+
+    const names = getAPINames();
+
+    const genericTuple = "<" + keyType + "," + valueType + ">";
+    const channelType = names.channel + genericTuple;
+    const productType = names.product + genericTuple;
+    const init = "KadabraChannel.newInstance(" + capacity + ")";
+    const mods = ["private"];
+    if (isStatic) {
+        mods.push("static");
+    }
+    const $channeltemp = $class.newField(mods, channelType, channelName, init);
+    const $channel = $channeltemp;
```

Comment:
Dá para contrair estas duas linhas
```suggestion
    const $channel = $class.newField(mods, channelType, channelName, init);
```
Não sei se isto foi uma das razões da confusão dos merges, mas a sugestão não acrescenta os ficheiros compilados, tens que fazer npm run build